lib/txnManager.js

```diff
@@ -0,0 +1,89 @@
+const assert = require('bsert');
+
+const TXUX = require('./uxtx.js');
+
+class TxnManager {
+  constructor(options) {
+    // hash -> parsed transaction
+    this._parsed = {};
+
+    if (options)
+      this.fromOptions(options);
+  }
+
+  static fromOptions(options) {
+    return new this().fromOptions(options);
+  }
+
+  fromOptions(options) {
+    if (options.custom) {
+      assert(typeof options.custom === 'function');
+      this.custom = options.custom;
+    }
+
+    if (options.constants) {
+      // TODO: validation
+      this.constants = options.constants;
```

Comment:
What are the constants for? Might be good to have something a little more descriptive.